lib/gax.js

```diff
@@ -255,6 +260,49 @@ function createBackoffSettings(initialRetryDelayMillis,
 exports.createBackoffSettings = createBackoffSettings;
 
 /**
+ * Parameters to the exponential backoff algorithm for retrying.
+ *
+ * @param {number} initialRetryDelayMillis - the initial delay time,
+ *   in milliseconds, between the completion of the first failed request and the
+ *   initiation of the first retrying request.
+ * @param {number} retryDelayMultiplier - the multiplier by which to
+ *   increase the delay time between the completion of failed requests, and the
+ *   initiation of the subsequent retrying request.
+ * @param {number} maxRetryDelayMillis - the maximum delay time, in
+ *   milliseconds, between requests. When this value is reached,
+ *   ``retryDelayMultiplier`` will no longer be used to increase delay time.
+ * @param {number} initialRpcTimeoutMillis - the initial timeout parameter
+ *   to the request.
+ * @param {number} rpcTimeoutMultiplier - the multiplier by which to
+ *   increase the timeout parameter between failed requests.
+ * @param {number} maxRpcTimeoutMillis - the maximum timeout parameter, in
+ *   milliseconds, for a request. When this value is reached,
+ *   ``rpcTimeoutMultiplier`` will no longer be used to increase the timeout.
+ * @property {number} maxRetries - the maximum number of retrying attempts that
+ *   will be made. If reached, an error will be returned.
+ * @return {BackoffSettings} a new settings.
+ *
+ */
+function createMaxRetriesBackoffSettings(initialRetryDelayMillis,
+                                         retryDelayMultiplier,
+                                         maxRetryDelayMillis,
+                                         initialRpcTimeoutMillis,
+                                         rpcTimeoutMultiplier,
+                                         maxRpcTimeoutMillis,
+                                         maxRetries) {
```

Comment:
this function is called from `constructSettings` below, and I believe `maxRetries` won't come from the JSON config. `maxRetries` should be at least optional.